src/Site.js

```diff
@@ -507,6 +518,28 @@ Site.prototype._rebuildAffectedSourceFiles = function (filePaths) {
 Site.prototype.rebuildAffectedSourceFiles
   = delay(Site.prototype._rebuildAffectedSourceFiles, 1000);
 
+Site.prototype._rebuildSourceFiles = function () {
+  logger.verbose('Rebuild all files');
+  return new Promise((resolve, reject) => {
+    Promise.resolve('')
+      .then(() => this.updateAddressablePages())
+      .then(filesToRemove => this.removeAsset(filesToRemove))
+      .then(() => this.buildSourceFiles())
```

Comment:
> There might be a way in which we only rebuild files that have been added which reduces build time. I'm not sure if it's worth the effort to implement though as there might be a lot of edge cases.

This might affect the build time of large websites though, like for example se-book. 

We could list the new files created (using the method that we used to find deleted files, but in reverse), and scan for all the includes of this file, then recreate those files? `regenerateAffectedPages()` seems to be using around the same logic.

Reply:
This is a little tricky... I tried making it work but am having trouble trying to get it to only regenerate certain pages properly. Will get back to it in future.